apps/module-scan/src/cli/retry-scan/index.ts

```diff
@@ -144,38 +161,66 @@ export async function retryScan(
         const originalScans: PageScan[] = [
           {
             interpretation: frontInterpretation,
-            originalFilename: frontOriginalFilename,
-            normalizedFilename: frontNormalizedFilename,
+            originalFilename: absolutify(frontOriginalFilename),
+            normalizedFilename: absolutify(frontNormalizedFilename),
           },
           {
             interpretation: backInterpretation,
-            originalFilename: backOriginalFilename,
-            normalizedFilename: backNormalizedFilename,
+            originalFilename: absolutify(backOriginalFilename),
+            normalizedFilename: absolutify(backNormalizedFilename),
           },
         ]
 
+        const [
+          frontDetectQrcodeOutput,
+          backDetectQrcodeOutput,
+        ] = await Promise.all(
+          originalScans.map(
+            async (scan) =>
+              (await pool.call({
```

Comment:
this may be not fully understanding async/await, but if you `await` in the mapped function, does it actually parallelize it?

Reply:
Yes, it does. `await` doesn't actually pause the JS runtime, it just allows it to switch to another "not a thread". See [this example](https://www.typescriptlang.org/play?ssl=1&ssc=1&pln=15&pc=7#code/IYZwngdgxgBAZgV2gFwJYHsIwLbFRACgEoYBvAKBipikxHQBsBTAOgfQHMCByEZYAE5oIHGACt0AIxDcilasADueZDAAKA9NlQhWwBgwIBtAEwAaGAEYAui1wAHAqEiwC2ECQC8APnnUFyqiqEEyK6praugA8AG7oqAAm3gQEAkz0DDFMXt5kfv4FtBAZrOxcAAZ8gsKiEpIwyAAWwKr8ANbpMAAkpO4Avu7lcgUjMLrIACqo2EzoCMgpOXmjKzR0jKWcBOV1Dc2t6Oht3b0gAyDw+DqNTAlD+avUaRlZxA+jfRbuwx8-VER-NbFDZsLbcSToJriKQXOBXEA3BKych9cjkXD4YhAA):

```ts
async function main() {
    console.log('starting jobs')
    await Promise.all([2, 1].map(async (ms) =>
        await new Promise<void>((resolve) => {
            console.log(`starting job that takes ${ms}ms`)
            setTimeout(() => {
                console.log(`job that took ${ms}ms finished`)
                resolve()
            }, ms)
        })
    ))
    console.log('both jobs finished')
}

main()
```

If you run it, you get this output:

```
[LOG]: "starting jobs" 
[LOG]: "starting job that takes 2ms" 
[LOG]: "starting job that takes 1ms" 
[LOG]: "job that took 2ms finished" 
[LOG]: "job that took 1ms finished" 
[LOG]: "both jobs finished" 
```

It is essentially equivalent to this, if thinking of it this way helps:

```ts
function main() {
  console.log('starting jobs')
  return Promise.all([2, 1].map((ms) =>
      new Promise<void>((resolve) => {
          console.log(`starting job that takes ${ms}ms`)
          setTimeout(() => {
              console.log(`job that took ${ms}ms finished`)
              resolve()
          }, ms)
      })
  )).then(() => {
    console.log('both jobs finished')
  })
}

main()
```